### PR TITLE
Add support for tagging AWS resources

### LINKED ISSUE #106: Add support for tagging AWS resources
I've been using Terratest in anger recently and have spotted a few things about the library.

Part of a pipeline that is building an AMI, I've found it quite common to add a tag to an AMI once it has passed testing. Adding a method to add tags to an AMI, along with other AWS resources, would be useful.

### PR DESCRIPTION
**Pull request by wjam**
Fixes #106

### CODE REVIEW

**Comment by wjam** [comment]
Output of test run: https://gist.github.com/wjam/ed2f6caf121082ed8d340d4a0220bb86

File: modules/aws/ec2.go [hunk 1]
```diff
@@ -164,6 +164,35 @@ func DeleteAmiE(t *testing.T, region string, imageID string) error {
...
+	client, err := NewEc2ClientE(t, region)
+	if err != nil {
+		return err
+	}
+
+	var awsTags []*ec2.Tag
+	for key, value := range tags {
+		awsTags = append(awsTags, &ec2.Tag{
+			Key: &key,
+			Value: &value,
```
Comment (brikis98) [inline modules/aws/ec2.go]: Use `aws.String(key)` and `aws.String(value)`

File: modules/aws/ec2.go [hunk 2]
```diff
@@ -164,6 +164,35 @@ func DeleteAmiE(t *testing.T, region string, imageID string) error {
...
 }
 
+func AddTagsToResource(t *testing.T, region string, resource string, tags map[string]string) {
+	err := AddTagsToResourceE(t, region, resource, tags)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func AddTagsToResourceE(t *testing.T, region string, resource string, tags map[string]string) error {
```
Comment (brikis98) [inline modules/aws/ec2.go]: Could you add comments to these methods specifying what types of resources are supported and what the `resource` param should be?

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.9.11

**Pull request merged**